### ISSUE TITLE
Fix Sql Test failures due to testFramework differences

### ISSUE THREAD

**Issue by markcowl** [Azure/azure-powershell#12276]
## Description
Set these tests for SKIP, because the ConfirmImpact setting was not validating correctly:
SqlVirtualMachine tests

Microsoft.Azure.Commands.SqlVirtualMachine.Test.ScenarioTests.UnitTest.AzureSqlVMGroupTests.GetAzureSqlVMGroup Microsoft.Azure.Commands.SqlVirtualMachine.Test.ScenarioTests.UnitTest.AzureSqlVMTests.GetAzureSqlVM 

SQl tests

 Microsoft.Azure.Commands.Sql.Test.UnitTests.AzureSqlDatabaseIndexRecommendationAttributeTests.StartAzureSqlDatabaseExecuteIndexRecommendationAttributes 
 Microsoft.Azure.Commands.Sql.Test.UnitTests.AzureSqlDatabaseIndexRecommendationAttributeTests.StopAzureSqlDatabaseExecuteIndexRecommendationAttributes 

Need to correct the issue and re-enable the tests

**Comment by VeryEarly** [comment]
Fix could be https://github.com/Azure/azure-powershell/pull/10586/files if we decide downgrade test framework to netcore 2.1

**Comment by markcowl** [comment]
Yeah, I think that makes sense (I assume you mean conditioning the assertion on `SupportsShouldProcess`).  This should work across PowerShell Core / NetCore versions.

I think we should spend some time this sprint or next adding netcoreapp3.1 as the other test framework, so we would have

powershell-core
  netcoreapp2.1
  netcoreapp3.1

windows-powershell
   net4.x

We should also think about scheduling some time to at least experiment with testing PowerShell core 7.1.  In september, when PowerShell core 6.x goes out of service, we will want to remove netcoreapp2.1 and enable .net 5.
